batcher-ui/src/pages/Main.tsx

```diff
@@ -82,23 +84,98 @@ const Welcome: React.FC = () => {
         if (status === BatcherStatus.OPEN) {
           setOpenTime(jsonData.value.status.open);
         }
-
-        console.log('%cMain.tsx line:114 jsonData.value', 'color: #007acc;', jsonData.value);
         setOrderBook(jsonData.value.orderbook);
       }
     } catch (error) {
       console.log('Batcher error', error);
     }
   };
 
+  const getClearedPayoutOfUserAddress = async () => {
+    if (!userAddress) {
+      setSellSideAmount(0);
+      setBuySideAmount(0);
+      return;
+    }
+
+    const storage = await contractsService.getStorage({
+      address: contractAddress,
+      level: 0,
+      path: null,
+    });
+
+    const userOrderBookURI = bigMapsByIdUri + storage.user_orderbook + '/keys/' + userAddress;
+    const userOrderBookData = await fetch(userOrderBookURI, { method: 'GET' });
+    const userOrderBooks = await userOrderBookData.json();
+    if (!Array.isArray(userOrderBooks.value.open) || userOrderBooks.value.open.length == 0) {
+      return;
+    }
+
+    const openOrderBooks = userOrderBooks.value.open;
+
+    console.log(344, openOrderBooks);
+
+    let initialBuySideAmount = 0;
+    let initialSellSideAmount = 0;
+
+    for (var i = 0; i < openOrderBooks.length; i++) {
+      const batchURI =
+        bigMapsByIdUri + storage.batch_set.batches + '/keys/' + openOrderBooks.at(i).batch_number;
+      const batchData = await fetch(batchURI, { method: 'GET' });
+      const batch = await batchData.json();
+      console.log(1444, batch);
+
+      const clearingRate =
+        parseInt(batch.value.status.cleared.rate.rate.val) *
+        10 ** parseInt(batch.value.status.cleared.rate.rate.pow);
+
+      const clearingKey = Object.keys(batch.value.status.cleared.clearing.clearing_tolerance)[0];
+      const clearing = parseInt(
+        batch.value.status.cleared.clearing.clearing_volumes[clearingKey.toLowerCase()],
+      );
+
+      console.log(2222, clearingRate);
+
+      const buySideActualVolume = parseInt(
+        batch.value.status.cleared.clearing.prorata_equivalence.buy_side_actual_volume,
+      );
+      const sellSideActualVolume = parseInt(
+        batch.value.status.cleared.clearing.prorata_equivalence.sell_side_actual_volume,
+      );
+
+      if (Object.keys(openOrderBooks.at(i).side)[0] === 'bUY') {
+        const depositedBuySideAmount = parseInt(openOrderBooks.at(i).swap.from.amount);
+        const unconvertedBuySideAmount =
```

Comment:
Not sure this is quite correct. 
1. We want to calculate how much (% or ratio) this order was in terms of all orders placed.  I.e this order was 10% of the whole buy volume placed in that batch.
2. We then know how much of the CLEARED volume that person should get.  I.e. if the cleared volume on the sell side was 10 mil USDT then they should get 10% of 10 mil USDT. 
3. We then need to work out how much of the buy side they have remaining.   (10% of 10 mil USDT) converted to tzBTC with the clearing price should always be LESS or EQUAL TO their original order amount in tzBTC.  
4. If LESS then it is a partial fill and then get their 10% of 10 mil USDT PLUS whatever they have remaining on the tzBTC side afer we have subtracted the equivalent in tzBTC from their order deposit amount. 
5. If EQUAL, then it is a COMPLETE fill and they just get their 10% of 10 mil USDT.   This however is less likely than a partial fill.

Reply:
Yup. I computed the exchanged tokens and remaining tokens based on the ratio. You can see it again in my new commits.